foundation/foundation-mda/src/test/java/com/boozallen/aiops/mda/generator/SparkApplicationStep.java

```diff
@@ -0,0 +1,134 @@
+package com.boozallen.aiops.mda.generator;/*-
+ * #%L
+ * aiSSEMBLE::Foundation::MDA
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aiops.mda.generator.util.PipelineUtils;
+
+import com.boozallen.aiops.mda.metamodel.element.*;
+import io.cucumber.java.Before;
+import io.cucumber.java.Scenario;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.fermenter.mda.GenerateSourcesHelper;
+import org.technologybrewery.fermenter.mda.element.ExpandedProfile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+public class SparkApplicationStep extends AbstractModelInstanceSteps {
+    private static final Logger logger = LoggerFactory.getLogger(SparkApplicationStep.class);
+
+    @Before("@spark-application-generation")
+    public void setup(Scenario scenario) throws IOException {
+        this.scenario = scenario.getName();
+        FileUtils.deleteDirectory(GENERATED_METADATA_DIRECTORY);
+    }
+
+    @Given("a spark project named {string}")
+    public void a_spark_project_named(String projectName) throws IOException {
+        createProject(projectName, "shared");
+    }
+
+    @Given("a data-flow pipeline using data-delivery-spark")
+    public void a_dataflow_pipeline_using_data_delivery_spark() throws IOException {
+        a_pipeline_using_data_delivery("data-flow", "data-delivery-spark");
+    }
+
+    @Given("a dataflow pipeline using {string}")
+    public void a_pipeline_using_data_delivery(String typeName, String implName) throws IOException {
+        createAndSavePipeline("SparkPipeline", "data-flow", implName);
+    }
+
+    @Given("a file store named {string}")
+    public void a_file_store_named(String fileStoreName) throws IOException {
+        List<String> fileStoreNames = new ArrayList<>();
+        fileStoreNames.add(fileStoreName);
+        saveFileStores(createFileStores(fileStoreNames));
+    }
+
+    @Given("two file stores named {string} and {string}")
+    public void a_pipeline_using_named_and_a_file_store_named_and(String fileStoreOneName, String fileStoreTwoName) throws IOException {
```

Comment:
A; Should update method name to better match the Given statement

Reply:
refactored.